tests/60app-services/02ghost.pl

```diff
@@ -157,3 +157,35 @@
          })->SyTest::pass_on_done( "Creator received ghost's message" )
       })->then_done(1);
    };
+
+my $unregistered_as_user_localpart = "astest-02ghost-1";
+
+test "Ghost user must register before joining room",
+   requires => [ $main::AS_USER, local_user_and_room_fixtures() ],
+
+   check => sub {
+      my ( $as_user, undef, $room_id ) = @_;
+
+      do_request_json_for( $as_user,
+         method => "POST",
+         uri    => "/api/v1/rooms/$room_id/join",
+         params => {
+            user_id => "@".$unregistered_as_user_localpart.":localhost:".$HOMESERVER_PORTS[0],
```

Comment:
You can get this a better way by requiring `$HOMESERVER_INFO[0]`, unpacking it into `$info` and then using

    $info->server_name

which will give you a string looking like `localhost:$port`